chat/src/user-info/Login.jsx

```diff
@@ -0,0 +1,162 @@
+  const initialValues = { username: "", email: "", password: "" };
+  const [formValues, setFormValues] = useState(initialValues);
+  const [formErrors, setFormErrors] = useState({});
+  const [user, setUser] = useState({});
+  const [isSubmit, setIsSubmit] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  // set the user filled data values in formValues
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormValues({ ...formValues, [name]: value });
+  };
+
+  // get the user authorization from backend
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setIsSubmit(true);
+    const auth = getAuth();
+    signInWithEmailAndPassword(
+      auth,
+      formValues.email,
+      formValues.password,
+      formValues.username
+    )
+    // user logged in successfully, directed to profile page
+      .then((userCredential) => {
+        const user = userCredential.user.email;
+        setFormErrors({ registered: "Login successful!" });
+        toast.success("logging in");
+        onValue(ref(db, "/Users/Signup/"), (querySnapShot) => {
+          querySnapShot.forEach((snap) => {
+            if (snap.val().email === user) {
+              localStorage.setItem("Name", JSON.stringify(snap.val()));
+              window.location.href = "/profile";
+            }
+          });
+        });
+      })
+      // errors defined
+      .catch((error) => {
+        if (error.code === "auth/user-not-found") {
+          setFormErrors({ email: "invalid email!" });
+        } else if (error.code === "auth/wrong-password") {
+          setFormErrors({ password: "wrong password" });
+        } else if (error.code === "auth/network-request-failed") {
+          alert(
+            "network error! please make sure that you have a working network access"
+          );
+        }
+      });
+  };
+  
+  useEffect(() => {
+    if (formErrors.length === 0 && isSubmit) {
+      //
+    }
+  }, [formErrors]);
+
+  return (
+    <>
+      <Navbar
+        className="container-fluid nav-bar"
+      >
+        <Link
+          to="/"
+          className=" text-black text-decoration-none m-2"
+        >
+          <img src="logo192.png" alt="logo" className="nav-logo"/>
+        </Link>
+        <div className="nav-text">
+          <Link
+            to="/login"
+            className="nav-login mx-4"
+          >
+            Login
+          </Link>
+          <Link to="/register" className="nav-register mx-3">
+            Register
+          </Link>
+        </div>
+      </Navbar>
+      <div className="login-card-main">
+        <Card
+          className="container login-card-inner"
+        >
+          <Card.Body>
+            <Card.Title className="text-center pb-3">Login</Card.Title>
+            <Form className="container" onSubmit={handleSubmit}>
+              <Form.Group className="m-1 p-1">
+                <Form.Label>Email :</Form.Label>
+                <Form.Control
```

Comment:
It would be good if you can make a component of this <Form.Control> and use it everywhere in the project.